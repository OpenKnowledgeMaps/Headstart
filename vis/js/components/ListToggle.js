import { connect } from "react-redux";

import ListToggleTemplate from "../templates/ListToggle";
import { toggleList } from "../actions";

// no logic required

const mapStateToProps = (state) => ({
  show: state.list.show,
});

const mapDispatchToProps = (dispatch) => ({
  onClick: () => {
    // TODO remove warn
    console.warn("*** React element 'List' click event triggered ***");
    dispatch(toggleList());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(ListToggleTemplate);
