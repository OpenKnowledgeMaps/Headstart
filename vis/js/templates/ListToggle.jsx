import React from "react";

const ListToggle = ({ toggleLabel, docsNumber, docsNumberLabel, onClick }) => {
  return (
    // html template starts here
    <div id="show_hide_button" className="row" onClick={onClick}>
      <div className="col-xs-2">▼</div>
      <div className="col-xs-8" id="show_hide_button_label">
        <span id="show_hide_label">
          <span>
            {toggleLabel}{" "}
            <span id="list_item_banner">
              (<span id="list_item_count">{docsNumber}</span> {docsNumberLabel})
            </span>
          </span>
        </span>
      </div>
      <div className="col-xs-2 text-right">▼</div>
    </div>
    // html template ends here
  );
};

export default ListToggle;
