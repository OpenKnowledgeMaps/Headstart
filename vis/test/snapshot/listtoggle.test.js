import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";

import ListToggle from "../../js/components/ListToggle";

const mockStore = configureStore([]);

describe("List toggle component snapshot", () => {
  it("matches a snapshot", () => {
    const store = mockStore({
      zoom: false,
      // TODO test some real data
      data: [],
      list: {
        show: true,
        searchValue: "",
        filterValue: "",
        filterField: "",
      },
      localization: {
        hide_list: "Liste einklappen",
        items: "Dokumente",
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <ListToggle />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
