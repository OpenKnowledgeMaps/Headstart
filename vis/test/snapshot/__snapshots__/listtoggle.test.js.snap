// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List toggle component snapshot matches a snapshot 1`] = `
Array [
  <div
    className="col-xs-2"
    onClick={[Function]}
  >
    ▼
  </div>,
  <div
    className="col-xs-8"
    id="show_hide_button_label"
    onClick={[Function]}
  >
    <span
      id="show_hide_label"
    >
      <span>
        Liste einklappen
         
        <span
          id="list_item_banner"
        >
          (
          <span
            id="list_item_count"
          >
            0
          </span>
           
          Dokumente
          )
        </span>
      </span>
    </span>
  </div>,
  <div
    className="col-xs-2 text-right"
    onClick={[Function]}
  >
    ▼
  </div>,
]
`;
